bigbang/mailman.py

```diff
@@ -47,63 +48,6 @@ def __str__(self):
         return repr(self.value)
 
 
-class MissingDataException(Exception):
-    def __init__(self, value):
-        self.value = value
-
-    def __str__(self):
-        return repr(self.value)
-
-
-def load_data(
-    name: str, archive_dir: str = CONFIG.mail_path, mbox: bool = False
-):
-    """
-    Load the data associated with an archive name, given as a string.
-
-    Attempt to open {archives-directory}/NAME.csv as data.
-
-    Failing that, if the the name is a URL, it will try to derive
-    the list name from that URL and load the .csv again.
-
-    Parameters
-    ----------
-    name : str
-    archive_dir : str, default CONFIG.mail_path
-    mbox : bool, default False
-        If true, expects and opens an mbox file at this path
-
-    Returns
-    -------
-    data : pandas.DataFrame
-
-    """
-
-    if mbox:
-        return open_list_archives(name, archive_dir=archive_dir, mbox=True)
-
-    # a first pass at detecting if the string is a URL...
-    if not (name.startswith("http://") or name.startswith("https://")):
-        path = os.path.join(archive_dir, name + ".csv")
-
-        if os.path.exists(path):
-            data = pd.read_csv(path)
-            return data
-        else:
-            logging.warning("No data available at %s", path)
-    else:
-        path = os.path.join(archive_dir, get_list_name(name) + ".csv")
-
-        if os.path.exists(path):
-            data = pd.read_csv(path)
-            return data
-        else:
-            logging.warning(
-                "No data found for %s. Check directory name and whether archives have been collected.",
-                name,
-            )
-
-
 def collect_from_url(
```

Comment:
Not quite sure if this fits here, but if this module (or one of the modules it refers to) is contains a loop that can take a while to execute, we could also address #354